packages/x-date-pickers-pro/src/DesktopDateRangePicker/DesktopDateRangePicker.tsx

```diff
@@ -53,6 +53,7 @@ const DesktopDateRangePicker = React.forwardRef(function DesktopDateRangePicker<
 
   const props = {
     ...defaultizedProps,
+    closeOnSelect: defaultizedProps.closeOnSelect ?? true,
```

Comment:
Explicitly providing `true` default in case of `DesktopDateRangePicker` and `DatePicker`.